### ISSUE TITLE
Kernel always crashes when using Tkinter

### ISSUE THREAD

**Issue by rkingery** [spyder-ide/spyder#4860]
Every time I've tried to use Tkinter in Spyder the kernel has crashed on me and I've had to restart it. I try running the simple "hello world" program included under the Tkinter help and get the following message in my kernel upon crashing:

<details>
> An error ocurred while starting the kernel
2017󈚫󈛂 11:14:01.953 python[13427:4689927] ‑[NSApplication _setup:]: unrecognized selector sent to instance 0x7fc3ee980d20
2017󈚫󈛂 11:14:01.972 python[13427:4689927] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '‑[NSApplication _setup:]: unrecognized selector sent to instance 0x7fc3ee980d20'
*** First throw call stack:
(
0 CoreFoundation 0x00007fffd3e392cb __exceptionPreprocess + 171
1 libobjc.A.dylib 0x00007fffe8c4448d objc_exception_throw + 48
2 CoreFoundation 0x00007fffd3ebaf04 ‑[NSObject(NSObject) doesNotRecognizeSelector:] + 132
3 CoreFoundation 0x00007fffd3dab755 ___forwarding___ + 1061
4 CoreFoundation 0x00007fffd3dab2a8 _CF_forwarding_prep_0 + 120
5 Tk 0x0000000117684c02 TkpInit + 471
6 Tk 0x00000001176002a9 Tk_Init + 1794
7 _tkinter.so 0x00000001048e0e44 Tcl_AppInit + 84
8 _tkinter.so 0x00000001048dbab4 Tkinter_Create + 1140
9 libpython2.7.dylib 0x000000010440e1bd PyEval_EvalFrameEx + 30141
10 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
11 libpython2.7.dylib 0x000000010439143c function_call + 364
12 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
13 libpython2.7.dylib 0x00000001043785b6 instancemethod_call + 182
14 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
15 libpython2.7.dylib 0x000000010441280d PyEval_CallObjectWithKeywords + 93
16 libpython2.7.dylib 0x00000001043765c6 PyInstance_New + 134
17 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
18 libpython2.7.dylib 0x000000010440de8f PyEval_EvalFrameEx + 29327
19 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
20 libpython2.7.dylib 0x000000010439143c function_call + 364
21 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
22 libpython2.7.dylib 0x00000001043785b6 instancemethod_call + 182
23 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
24 libpython2.7.dylib 0x000000010440de8f PyEval_EvalFrameEx + 29327
25 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
26 libpython2.7.dylib 0x000000010439143c function_call + 364
27 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
28 libpython2.7.dylib 0x00000001043785b6 instancemethod_call + 182
29 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
30 libpython2.7.dylib 0x000000010440de8f PyEval_EvalFrameEx + 29327
31 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
32 libpython2.7.dylib 0x000000010439143c function_call + 364
33 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
34 libpython2.7.dylib 0x00000001043785b6 instancemethod_call + 182
35 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
36 libpython2.7.dylib 0x000000010440de8f PyEval_EvalFrameEx + 29327
37 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
38 libpython2.7.dylib 0x000000010439143c function_call + 364
39 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
40 libpython2.7.dylib 0x00000001043785b6 instancemethod_call + 182
41 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
42 libpython2.7.dylib 0x000000010441280d PyEval_CallObjectWithKeywords + 93
43 libpython2.7.dylib 0x00000001043765c6 PyInstance_New + 134
44 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
45 libpython2.7.dylib 0x000000010440de8f PyEval_EvalFrameEx + 29327
46 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
47 libpython2.7.dylib 0x00000001044062e6 PyEval_EvalCode + 54
48 libpython2.7.dylib 0x0000000104435254 PyRun_FileExFlags + 164
49 libpython2.7.dylib 0x000000010440284b builtin_execfile + 507
50 libpython2.7.dylib 0x000000010440e9d7 PyEval_EvalFrameEx + 32215
51 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
52 libpython2.7.dylib 0x0000000104413015 fast_function + 117
53 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
54 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
55 libpython2.7.dylib 0x0000000104413015 fast_function + 117
56 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
57 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
58 libpython2.7.dylib 0x0000000104410871 PyEval_EvalFrameEx + 40049
59 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
60 libpython2.7.dylib 0x0000000104413015 fast_function + 117
61 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
62 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
63 libpython2.7.dylib 0x0000000104413015 fast_function + 117
64 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
65 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
66 libpython2.7.dylib 0x000000010439143c function_call + 364
67 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
68 libpython2.7.dylib 0x000000010440e901 PyEval_EvalFrameEx + 32001
69 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
70 libpython2.7.dylib 0x0000000104413015 fast_function + 117
71 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
72 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
73 libpython2.7.dylib 0x0000000104413015 fast_function + 117
74 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
75 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
76 libpython2.7.dylib 0x0000000104413015 fast_function + 117
77 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
78 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
79 libpython2.7.dylib 0x0000000104413015 fast_function + 117
80 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
81 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
82 libpython2.7.dylib 0x000000010439143c function_call + 364
83 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
84 libpython2.7.dylib 0x000000010440e901 PyEval_EvalFrameEx + 32001
85 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
86 libpython2.7.dylib 0x000000010439143c function_call + 364
87 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
88 libpython2.7.dylib 0x000000010440e901 PyEval_EvalFrameEx + 32001
89 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
90 libpython2.7.dylib 0x0000000104413015 fast_function + 117
91 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
92 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
93 libpython2.7.dylib 0x0000000104413015 fast_function + 117
94 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
95 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
96 libpython2.7.dylib 0x000000010439143c function_call + 364
97 libpython2.7.dylib 0x000000010436b333 PyObject_Call + 99
98 libpython2.7.dylib 0x000000010440e901 PyEval_EvalFrameEx + 32001
99 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
100 libpython2.7.dylib 0x0000000104413015 fast_function + 117
101 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
102 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
103 libpython2.7.dylib 0x0000000104413015 fast_function + 117
104 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
105 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
106 libpython2.7.dylib 0x0000000104413015 fast_function + 117
107 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
108 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
109 libpython2.7.dylib 0x0000000104413015 fast_function + 117
110 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
111 libpython2.7.dylib 0x00000001044130b2 fast_function + 274
112 libpython2.7.dylib 0x000000010440ddb3 PyEval_EvalFrameEx + 29107
113 libpython2.7.dylib 0x00000001044068f2 PyEval_EvalCodeEx + 1538
114 libpython2.7.dylib 0x00000001044062e6 PyEval_EvalCode + 54
115 libpython2.7.dylib 0x0000000104435254 PyRun_FileExFlags + 164
116 libpython2.7.dylib 0x0000000104434dd1 PyRun_SimpleFileExFlags + 769
117 libpython2.7.dylib 0x000000010444aa12 Py_Main + 3154
118 python 0x0000000104357f24 start + 52
)
libc++abi.dylib: terminating with uncaught exception of type NSException
2017󈚫󈛂 11:15:39.703 python[66577:5167715] ‑[NSApplication _setup:]: unrecognized selector sent to instance 0x7fce65eac0e0
2017󈚫󈛂 11:15:39.708 python[66577:5167715] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '‑[NSApplication _setup:]: unrecognized selector sent to instance 0x7fce65eac0e0'
*** First throw call stack:
(
0 CoreFoundation 0x00007fffd3e392cb __exceptionPreprocess + 171
1 libobjc.A.dylib 0x00007fffe8c4448d objc_exception_throw + 48
2 CoreFoundation 0x00007fffd3ebaf04 ‑[NSObject(NSObject) doesNotRecognizeSelector:] + 132
3 CoreFoundation 0x00007fffd3dab755 ___forwarding___ + 1061
4 CoreFoundation 0x00007fffd3dab2a8 _CF_forwarding_prep_0 + 120
5 Tk 0x000000011aa79c02 TkpInit + 471
6 Tk 0x000000011a9f52a9 Tk_Init + 1794
7 _tkinter.so 0x00000001179eae44 Tcl_AppInit + 84
8 _tkinter.so 0x00000001179e5ab4 Tkinter_Create + 1140
9 libpython2.7.dylib 0x000000010aa8a1bd PyEval_EvalFrameEx + 30141
10 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
11 libpython2.7.dylib 0x000000010aa0d43c function_call + 364
12 libpython2.7.dylib 0x000000010a9e7333 PyObject_Call + 99
13 libpython2.7.dylib 0x000000010a9f45b6 instancemethod_call + 182
14 libpython2.7.dylib 0x000000010a9e7333 PyObject_Call + 99
15 libpython2.7.dylib 0x000000010aa8e80d PyEval_CallObjectWithKeywords + 93
16 libpython2.7.dylib 0x000000010a9f25c6 PyInstance_New + 134
17 libpython2.7.dylib 0x000000010a9e7333 PyObject_Call + 99
18 libpython2.7.dylib 0x000000010aa89e8f PyEval_EvalFrameEx + 29327
19 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
20 libpython2.7.dylib 0x000000010aa822e6 PyEval_EvalCode + 54
21 libpython2.7.dylib 0x000000010aab1254 PyRun_FileExFlags + 164
22 libpython2.7.dylib 0x000000010aa7e84b builtin_execfile + 507
23 libpython2.7.dylib 0x000000010aa8a9d7 PyEval_EvalFrameEx + 32215
24 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
25 libpython2.7.dylib 0x000000010aa8f015 fast_function + 117
26 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
27 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
28 libpython2.7.dylib 0x000000010aa8f015 fast_function + 117
29 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
30 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
31 libpython2.7.dylib 0x000000010aa8c871 PyEval_EvalFrameEx + 40049
32 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
33 libpython2.7.dylib 0x000000010aa8f015 fast_function + 117
34 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
35 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
36 libpython2.7.dylib 0x000000010aa8f015 fast_function + 117
37 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
38 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
39 libpython2.7.dylib 0x000000010aa0d43c function_call + 364
40 libpython2.7.dylib 0x000000010a9e7333 PyObject_Call + 99
41 libpython2.7.dylib 0x000000010aa8a901 PyEval_EvalFrameEx + 32001
42 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
43 libpython2.7.dylib 0x000000010aa8f015 fast_function + 117
44 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
45 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
46 libpython2.7.dylib 0x000000010aa8f015 fast_function + 117
47 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
48 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
49 libpython2.7.dylib 0x000000010aa8f015 fast_function + 117
50 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
51 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
52 libpython2.7.dylib 0x000000010aa8f015 fast_function + 117
53 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
54 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
55 libpython2.7.dylib 0x000000010aa0d43c function_call + 364
56 libpython2.7.dylib 0x000000010a9e7333 PyObject_Call + 99
57 libpython2.7.dylib 0x000000010aa8a901 PyEval_EvalFrameEx + 32001
58 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
59 libpython2.7.dylib 0x000000010aa0d43c function_call + 364
60 libpython2.7.dylib 0x000000010a9e7333 PyObject_Call + 99
61 libpython2.7.dylib 0x000000010aa8a901 PyEval_EvalFrameEx + 32001
62 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
63 libpython2.7.dylib 0x000000010aa8f015 fast_function + 117
64 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
65 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
66 libpython2.7.dylib 0x000000010aa8f015 fast_function + 117
67 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
68 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
69 libpython2.7.dylib 0x000000010aa0d43c function_call + 364
70 libpython2.7.dylib 0x000000010a9e7333 PyObject_Call + 99
71 libpython2.7.dylib 0x000000010aa8a901 PyEval_EvalFrameEx + 32001
72 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
73 libpython2.7.dylib 0x000000010aa8f015 fast_function + 117
74 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
75 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
76 libpython2.7.dylib 0x000000010aa8f015 fast_function + 117
77 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
78 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
79 libpython2.7.dylib 0x000000010aa8f015 fast_function + 117
80 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
81 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
82 libpython2.7.dylib 0x000000010aa8f015 fast_function + 117
83 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
84 libpython2.7.dylib 0x000000010aa8f0b2 fast_function + 274
85 libpython2.7.dylib 0x000000010aa89db3 PyEval_EvalFrameEx + 29107
86 libpython2.7.dylib 0x000000010aa828f2 PyEval_EvalCodeEx + 1538
87 libpython2.7.dylib 0x000000010aa822e6 PyEval_EvalCode + 54
88 libpython2.7.dylib 0x000000010aab1254 PyRun_FileExFlags + 164
89 libpython2.7.dylib 0x000000010aab0dd1 PyRun_SimpleFileExFlags + 769
90 libpython2.7.dylib 0x000000010aac6a12 Py_Main + 3154
91 python 0x000000010a9d3f24 start + 52
)
libc++abi.dylib: terminating with uncaught exception of type NSException
2017󈚫󈛂 11:22:04.659 python[66785:5170781] ‑[NSApplication _setup:]: unrecognized selector sent to instance 0x7f89900da090
2017󈚫󈛂 11:22:04.665 python[66785:5170781] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '‑[NSApplication _setup:]: unrecognized selector sent to instance 0x7f89900da090'
*** First throw call stack:
(
0 CoreFoundation 0x00007fffd3e392cb __exceptionPreprocess + 171
1 libobjc.A.dylib 0x00007fffe8c4448d objc_exception_throw + 48
2 CoreFoundation 0x00007fffd3ebaf04 ‑[NSObject(NSObject) doesNotRecognizeSelector:] + 132
3 CoreFoundation 0x00007fffd3dab755 ___forwarding___ + 1061
4 CoreFoundation 0x00007fffd3dab2a8 _CF_forwarding_prep_0 + 120
5 Tk 0x000000011e1f8c02 TkpInit + 471
6 Tk 0x000000011e1742a9 Tk_Init + 1794
7 _tkinter.so 0x000000011e04de44 Tcl_AppInit + 84
8 _tkinter.so 0x000000011e048ab4 Tkinter_Create + 1140
9 libpython2.7.dylib 0x000000010e21a1bd PyEval_EvalFrameEx + 30141
10 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
11 libpython2.7.dylib 0x000000010e19d43c function_call + 364
12 libpython2.7.dylib 0x000000010e177333 PyObject_Call + 99
13 libpython2.7.dylib 0x000000010e1845b6 instancemethod_call + 182
14 libpython2.7.dylib 0x000000010e177333 PyObject_Call + 99
15 libpython2.7.dylib 0x000000010e21e80d PyEval_CallObjectWithKeywords + 93
16 libpython2.7.dylib 0x000000010e1825c6 PyInstance_New + 134
17 libpython2.7.dylib 0x000000010e177333 PyObject_Call + 99
18 libpython2.7.dylib 0x000000010e219e8f PyEval_EvalFrameEx + 29327
19 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
20 libpython2.7.dylib 0x000000010e2122e6 PyEval_EvalCode + 54
21 libpython2.7.dylib 0x000000010e241254 PyRun_FileExFlags + 164
22 libpython2.7.dylib 0x000000010e20e84b builtin_execfile + 507
23 libpython2.7.dylib 0x000000010e21a9d7 PyEval_EvalFrameEx + 32215
24 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
25 libpython2.7.dylib 0x000000010e21f015 fast_function + 117
26 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
27 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
28 libpython2.7.dylib 0x000000010e21f015 fast_function + 117
29 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
30 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
31 libpython2.7.dylib 0x000000010e21c871 PyEval_EvalFrameEx + 40049
32 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
33 libpython2.7.dylib 0x000000010e21f015 fast_function + 117
34 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
35 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
36 libpython2.7.dylib 0x000000010e21f015 fast_function + 117
37 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
38 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
39 libpython2.7.dylib 0x000000010e19d43c function_call + 364
40 libpython2.7.dylib 0x000000010e177333 PyObject_Call + 99
41 libpython2.7.dylib 0x000000010e21a901 PyEval_EvalFrameEx + 32001
42 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
43 libpython2.7.dylib 0x000000010e21f015 fast_function + 117
44 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
45 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
46 libpython2.7.dylib 0x000000010e21f015 fast_function + 117
47 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
48 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
49 libpython2.7.dylib 0x000000010e21f015 fast_function + 117
50 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
51 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
52 libpython2.7.dylib 0x000000010e21f015 fast_function + 117
53 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
54 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
55 libpython2.7.dylib 0x000000010e19d43c function_call + 364
56 libpython2.7.dylib 0x000000010e177333 PyObject_Call + 99
57 libpython2.7.dylib 0x000000010e21a901 PyEval_EvalFrameEx + 32001
58 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
59 libpython2.7.dylib 0x000000010e19d43c function_call + 364
60 libpython2.7.dylib 0x000000010e177333 PyObject_Call + 99
61 libpython2.7.dylib 0x000000010e21a901 PyEval_EvalFrameEx + 32001
62 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
63 libpython2.7.dylib 0x000000010e21f015 fast_function + 117
64 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
65 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
66 libpython2.7.dylib 0x000000010e21f015 fast_function + 117
67 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
68 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
69 libpython2.7.dylib 0x000000010e19d43c function_call + 364
70 libpython2.7.dylib 0x000000010e177333 PyObject_Call + 99
71 libpython2.7.dylib 0x000000010e21a901 PyEval_EvalFrameEx + 32001
72 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
73 libpython2.7.dylib 0x000000010e21f015 fast_function + 117
74 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
75 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
76 libpython2.7.dylib 0x000000010e21f015 fast_function + 117
77 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
78 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
79 libpython2.7.dylib 0x000000010e21f015 fast_function + 117
80 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
81 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
82 libpython2.7.dylib 0x000000010e21f015 fast_function + 117
83 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
84 libpython2.7.dylib 0x000000010e21f0b2 fast_function + 274
85 libpython2.7.dylib 0x000000010e219db3 PyEval_EvalFrameEx + 29107
86 libpython2.7.dylib 0x000000010e2128f2 PyEval_EvalCodeEx + 1538
87 libpython2.7.dylib 0x000000010e2122e6 PyEval_EvalCode + 54
88 libpython2.7.dylib 0x000000010e241254 PyRun_FileExFlags + 164
89 libpython2.7.dylib 0x000000010e240dd1 PyRun_SimpleFileExFlags + 769
90 libpython2.7.dylib 0x000000010e256a12 Py_Main + 3154
91 python 0x000000010e163f24 start + 52
)
libc++abi.dylib: terminating with uncaught exception of type NSException
</details>
<br>
I only have this problem when trying to run such programs in Spyder.  They run fine in Jupyter, qt console, and in the command line python.  I also tried downgrading to previous versions of Spyder and got the same issues.

**What steps will reproduce the problem?**

1. If this is a bug, please fix.
2. If this is not a bug, please point me to a fix if possible.

**What is the expected output? What do you see instead?**

I see error messages and my kernel crashes.  That's it.

**Please provide any additional information below**

I run Spyder via Anaconda.

## Versions and main components

* Spyder Version: 3.2.0 (tried on 3.1.4 as well and got problems too)
* Python Version: 2.7
* Qt Version:  4.3.0
* PyQt Version:
* Operating system: macOS Sierra

## Dependencies

Please go to the menu entry `Help > Optional Dependencies` (or
`Help > Dependencies`), press the button `Copy to clipboard`
and paste the contents below:

IPython >=4.0;<6.0:  5.3.0 (OK)
cython >=0.21     :  0.25.2 (OK)
jedi >=0.9.0      :  0.10.2 (OK)
nbconvert >=4.0   :  5.1.1 (OK)
numpy >=1.7       :  1.12.1 (OK)
pandas >=0.13.1   :  0.20.1 (OK)
psutil >=0.3      :  5.2.2 (OK)
pycodestyle >=2.3 :  2.3.1 (OK)
pyflakes >=0.5.0  :  1.5.0 (OK)
pygments >=2.0    :  2.2.0 (OK)
pylint >=0.25     :  1.6.4 (OK)
qtconsole >=4.2.0 :  4.3.0 (OK)
rope >=0.9.4      :  0.9.4 (OK)
sphinx >=0.6.6    :  1.5.6 (OK)
sympy >=0.7.3     :  1.0 (OK)

**Comment by ccordoba12** [comment]
Thanks for reporting. We'll take a look at it in a future release.

Question: are you running `%gui tk` before running your code in our consoles?

**Comment by rkingery** [comment]
No, but after just running %gui tk in the console I do get the same kind of
error as I've been getting when running the usual code.

**Comment by armandotoledo** [comment]
I have the same problem with Spyder 3.2 and Mac OS Sierra

**Comment by goanpeca** [comment]
Just tried this simple code

```
from tkinter import *
window = Tk()
window.title("Welcome to LikeGeeks app")
lbl = Label(window, text="Hello")
lbl.grid(column=0, row=0)
window.mainloop()
```

And works as expected.
<img width="458" alt="Screen Shot 2019-12-03 at 22 03 30" src="https://user-images.githubusercontent.com/3627835/70109041-c34f4600-1618-11ea-919e-e1d07af47d3a.png">

Closing

**Comment by JBum21** [comment]
I'm still getting this issue even with the example program you gave.

**Comment by jilanr** [comment]
I am also still getting this problem, including with @goanpeca's code - although I can run it through Jupyter and Terminal no problem.  It's Spyder specific.  I've updated my operating system, anaconda, and spyder to all the latest versions, and nothing seems to fix. 

Running through anaconda causes the kernel to restart, while running just through Spyder indirectly produces the error: 

ModuleNotFoundError: No module named '_tkinter'

I would easily believe that I'm doing something dumb (I'm pretty new to programming) but literally nothing I've found on any message board helps resolve this problem, so it's definitely still an issue for me :(
<img width="1792" alt="Screen Shot 2021-07-16 at 10 27 24 AM" src="https://user-images.githubusercontent.com/87536023/125966224-2571e499-b9f4-4538-90cb-35b94632da6a.png">
